worker/provisioner/provisioner_task.go

```diff
@@ -89,12 +89,16 @@ func NewProvisionerTask(
 ) (ProvisionerTask, error) {
 	machineChanges := machineWatcher.Changes()
```

Comment:
Should we also be adding in the machineWatcher?

Reply:
It seems weird that the task would take ownership of two of the three watchers passed in.

Reply:
So sorry, am blind.